File: 프로그래머스/unrated/181864. 문자열 바꿔서 찾기/문자열 바꿔서 찾기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(string myString, string pat) {
    for (int i = 0; i < myString.size(); i++)
    {
        if (myString[i] == 'A')
            myString[i] = 'B';
        else
            myString[i] = 'A';
    }
    
    return myString.find(pat) != string::npos ? 1 : 0;
}

File: 프로그래머스/0/181864. 문자열 바꿔서 찾기/문자열 바꿔서 찾기.java
class Solution {
    public int solution(String myString, String pat) {
        char[] arr = myString.toCharArray();
        for (int i = 0; i < arr.length; i++) arr[i] = arr[i] == 'A' ? 'B' : 'A';

        return String.valueOf(arr).contains(pat) ? 1 : 0;
    }
}